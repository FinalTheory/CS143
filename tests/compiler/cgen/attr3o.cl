--Test of attributes of type SELF_TYPE
(*Requires dispatch and new to work*)

class Alpha
{
   s:SELF_TYPE <- self;
   getNum() : Int {5};
};

class Beta inherits Alpha
{
   i:Int <- 10;
   getSelf() : SELF_TYPE {s};
   getNum() : Int {i};
};

class Main inherits IO
{
   b:Beta <- new Beta;
   main() : Object {{
     out_int(b.getSelf().getNum());
     self;
   }};
};
