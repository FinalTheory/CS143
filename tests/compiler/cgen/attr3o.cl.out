SPIM Version 6.5 of January 4, 2003
Copyright 1990-2003 by James R. Larus (larus@cs.wisc.edu).
All Rights Reserved.
See the file README for a full copyright notice.
Loaded: /usr/class/cs143/cool/lib/trap.handler
10COOL program successfully executed
Stats -- #instructions : 295
         #reads : 62  #writes 49  #branches 56  #other 128
